### ISSUE TITLE
[PDF] Unable to recognise local paths

### ISSUE THREAD

**Issue by MZuhairKhan** [jupyter/nbconvert#1998]
TLDR: nbconvert appears to be unable to recognise local path. I tried to change a notebook to pdf and it says that an image does not exist.  It says ../images/errorcorrectionmodel.png does not exist when it does. VSCode was showing a link to the file in the error message.

Here is my directory structure:

.qecft
├── build
│   ├── cec
│   │   ├── cec.ipynb
│   │   ├── ....
│   ├──images
│   │   ├── errorcorrectionalmodel.png
│   │   ├── ...
│   ├── ....
|──images
│   ├── errorcorrectionalmodel.png
│   ├── ....

OS Name:                   Microsoft Windows 10 Home
OS Version:                10.0.19044 N/A Build 19044
MiKTeX-XeTeX 4.10 (MiKTeX 23.5)
jupyter nbconvert --version: 7.4.0

Here is the command and error log:
(envqecft) C:\Users\khanm\OneDrive\Desktop\Projects\qecft\build\cec>jupyter nbconvert --to pdf cec.ipynb
[NbConvertApp] Converting notebook cec.ipynb to pdf
C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbformat\__init__.py:93: MissingIDFieldWarning: Code cell is missing an id field, this will become a hard error in future nbformat versions. You may want to use `normalize()` on your notebooks before validations (available since nbformat 5.1.4). Previous versions of nbformat are fixing this issue transparently, and will stop doing so in the future.
  validate(nb)
[NbConvertApp] Writing 61774 bytes to notebook.tex
[NbConvertApp] Building PDF
[NbConvertApp] Running xelatex 3 times: ['xelatex', 'notebook.tex', '-quiet']
[NbConvertApp] CRITICAL | x failed: xelatex notebook.tex -quiet
b'\r\ndvipdfmx:fatal: Image inclusion failed. Could not find file: ../images/errorcorrectionmodel.png\r\n\r\nNo output PDF file written.\r\n\r\nSorry, but xelatex did not succeed.\r\n\r\nThe log file hopefully contains the information to get MiKTeX going again:\r\n\r\n  C:\\Users\\khanm\\AppData\\Local\\MiKTeX\\miktex\\log\\xelatex.log\r\n'
Traceback (most recent call last):
  File "C:\Users\khanm\miniconda3\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\khanm\miniconda3\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\Scripts\jupyter-nbconvert.EXE\__main__.py", line 7, in <module>
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\jupyter_core\application.py", line 277, in launch_instance
    return super().launch_instance(argv=argv, **kwargs)
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\traitlets\config\application.py", line 1043, in launch_instance
    app.start()
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\nbconvertapp.py", line 414, in start
    self.convert_notebooks()
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\nbconvertapp.py", line 589, in convert_notebooks
    self.convert_single_notebook(notebook_filename)
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\nbconvertapp.py", line 555, in convert_single_notebook
    output, resources = self.export_single_notebook(
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\nbconvertapp.py", line 481, in export_single_notebook
    output, resources = self.exporter.from_filename(
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\exporters\templateexporter.py", line 388, in from_filename
    return super().from_filename(filename, resources, **kw)  # type:ignore
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\exporters\exporter.py", line 201, in from_filename
    return self.from_file(f, resources=resources, **kw)
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\exporters\templateexporter.py", line 394, in from_file
    return super().from_file(file_stream, resources, **kw)  # type:ignore
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\exporters\exporter.py", line 220, in from_file
    return self.from_notebook_node(
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\exporters\pdf.py", line 199, in from_notebook_node
    self.run_latex(tex_file)
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\exporters\pdf.py", line 168, in run_latex
    return self.run_command(
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\exporters\pdf.py", line 158, in run_command
    raise raise_on_failure(msg)
nbconvert.exporters.pdf.LatexFailed: PDF creating failed, captured latex output:
Failed to run "xelatex notebook.tex -quiet" command:

dvipdfmx:fatal: Image inclusion failed. Could not find file: ../images/errorcorrectionmodel.png

No output PDF file written.

Sorry, but xelatex did not succeed.

The log file hopefully contains the information to get MiKTeX going again:

  C:\Users\khanm\AppData\Local\MiKTeX\miktex\log\xelatex.log

-------------------------------------------

MiKTeX logs:
2023-05-31 23:12:00,183+0300 INFO  xelatex - this process (6244) started by pandoc in directory C:\Users\khanm\OneDrive\Desktop\Projects\qecft with command line: xelatex -halt-on-error -interaction nonstopmode -output-directory C:/Users/khanm/AppData/Local/Temp/tex2pdf.-5bc1ca47d967cf83 C:/Users/khanm/AppData/Local/Temp/tex2pdf.-5bc1ca47d967cf83/input.tex
2023-05-31 23:12:00,184+0300 INFO  xelatex - running on Windows 10.0.19044

This is nowhere near the current time. 

Any ideas on how to fix this?

**Comment by MZuhairKhan** [comment]
Changing to local path creates the following error:
(envqecft) C:\Users\khanm\OneDrive\Desktop\Projects\qecft\build\cec>jupyter nbconvert --to pdf cec.ipynb
[NbConvertApp] Converting notebook cec.ipynb to pdf
C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbformat\__init__.py:93: MissingIDFieldWarning: Code cell is missing an id field, this will become a hard error in future nbformat versions. You may want to use `normalize()` on your notebooks before validations (available since nbformat 5.1.4). Previous versions of nbformat are fixing this issue transparently, and will stop doing so in the future.
  validate(nb)
[NbConvertApp] Writing 61874 bytes to notebook.tex
[NbConvertApp] Building PDF
[NbConvertApp] Running xelatex 3 times: ['xelatex', 'notebook.tex', '-quiet']
[NbConvertApp] CRITICAL | x failed: xelatex notebook.tex -quiet
b''
Traceback (most recent call last):
  File "C:\Users\khanm\miniconda3\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\khanm\miniconda3\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\Scripts\jupyter-nbconvert.EXE\__main__.py", line 7, in <module>
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\jupyter_core\application.py", line 277, in launch_instance
    return super().launch_instance(argv=argv, **kwargs)
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\traitlets\config\application.py", line 1043, in launch_instance
    app.start()
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\nbconvertapp.py", line 414, in start
    self.convert_notebooks()
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\nbconvertapp.py", line 589, in convert_notebooks
    self.convert_single_notebook(notebook_filename)
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\nbconvertapp.py", line 555, in convert_single_notebook
    output, resources = self.export_single_notebook(
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\nbconvertapp.py", line 481, in export_single_notebook
    output, resources = self.exporter.from_filename(
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\exporters\templateexporter.py", line 388, in from_filename
    return super().from_filename(filename, resources, **kw)  # type:ignore
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\exporters\exporter.py", line 201, in from_filename
    return self.from_file(f, resources=resources, **kw)
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\exporters\templateexporter.py", line 394, in from_file
    return super().from_file(file_stream, resources, **kw)  # type:ignore
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\exporters\exporter.py", line 220, in from_file
    return self.from_notebook_node(
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\exporters\pdf.py", line 199, in from_notebook_node
    self.run_latex(tex_file)
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\exporters\pdf.py", line 168, in run_latex
    return self.run_command(
  File "C:\Users\khanm\OneDrive\Desktop\Projects\qecft\envqecft\lib\site-packages\nbconvert\exporters\pdf.py", line 158, in run_command
    raise raise_on_failure(msg)
nbconvert.exporters.pdf.LatexFailed: PDF creating failed, captured latex output:
Failed to run "xelatex notebook.tex -quiet" command:

**Comment by MZuhairKhan** [comment]
Some additional notes which might help: I have combined multiple jupyter notebooks to create this file. Changing just 1 of the original jupyter notebooks is fine as long as I also switch the paths to absolute paths.

Here is my code to create the notebook: 

def combine_notebooks(notebook_paths, output_path):
    """
    Combine multiple Jupyter notebooks into a single one.

    :param notebook_paths: A list of file paths of the notebooks to combine.
    :type notebook_paths: List[str]
    :param output_path: The file path to save the combined notebook to.
    :type output_path: str

    :return: None
    """
    # Create a new notebook object
    combined_notebook = new_notebook()

    # For each notebook in the list
    for path in notebook_paths:
        # Read the notebook and parse it using nbformat
        with open(path, "r", encoding="utf-8") as f:
            notebook = nbformat.read(f, as_version=4)

        # For each cell in the notebook
        for cell in notebook.cells:
            # If the cell is a code cell, add it to the combined notebook as a new code cell
            if cell.cell_type == "code":
                combined_notebook.cells.append(new_code_cell(source=cell.source))
            # If the cell is a markdown cell, add it to the combined notebook as a new markdown cell
            elif cell.cell_type == "markdown":
                # Replace relative paths with absolute paths to fix images
                #cell.source = cell.source.replace('](../..', '](..')
                #output_dir = os.sep.join(output_path.split(os.sep)[:-2])
                #cell.source = cell.source.replace('](../..', '](' + output_dir)


                combined_notebook.cells.append(new_markdown_cell(source=cell.source))

    # Delete the output file if it exists
    delete_file(output_path)
    # Write the combined notebook to the output file
    with open(output_path, "w", encoding="utf-8") as f:
        nbformat.write(combined_notebook, f)

    # Print a message confirming that the combined notebook was saved
    print(f"Combined notebook saved to '{output_path}'.")

**Comment by EYH0602** [comment]
this also happens when trying to use local image. My directory structure is 
```
> tree .
.
├── LICENSE
├── README.md
├── data
│   ├── Flatten_squirrel.png
│   ├── student_assignment.csv
│   └── valueshist
│       ├── Developed_ex-US.csv
│       ├── Emerging_Markets.csv
│       ├── Europe.csv
│       ├── Global.csv
│       └── US.csv
├── hw4.ipynb
├── image
│   └── hw4_4A.png
└── requirements.txt
```

When converting `hw4.ipynb` to PDF, it would have error

```
LaTeX Warning: File `./image/hw4_4A.png' not found on input line 413.

! Unable to load picture or PDF file './image/hw4_4A.png'.
<to be read again> 
                   }
l.413 \includegraphics{./image/hw4_4A.png}
                                          
? 
! Emergency stop.
<to be read again> 
                   }
l.413 \includegraphics{./image/hw4_4A.png}
```

In the markdown part in `hw4.ipynb`, the image is included using

```md
![boutique coffee roasting company problem](./image/hw4_4A.png)
```

It would be no problem if this line is changed to

```md
![boutique coffee roasting company problem](image/hw4_4A.png)
```

---

version: 6.5.0
conda: 23.3.1
platform: Apple M1 Pro

**Comment by tuncbkose** [comment]
This is pretty weird, especially

> It would be no problem if this line is changed to ...

And I am pretty sure it works fine if you were to first convert to `.tex` via `nbconvert`, and `xelatex` on your own (someone please correct me if this is not the case).

So my guess is that something wonky is happening in how `nbconvert` is trying to forward the context of the document to the `latex` command. This is done via the `TEXINPUTS` environment variable, but that's all I know about it so far. Maybe paths with `./` or `../` doesn't work with `TEXINPUTS`.

**Comment by tuncbkose** [comment]
I can reproduce this behavior with the following:
```
$ mkdir d1
$ mkdir d1/d2
$ touch d1/d2/f.jpg
$ kpsewhich d2/f.jpg
$ TEXINPUTS=d1 kpsewhich d2/f.jpg
d1/d2/f.jpg
$ TEXINPUTS=d1 kpsewhich ./d2/f.jpg
```

**Comment by tuncbkose** [comment]
I will create a PR for this soon, need to fix a related issue first.

**Comment by tuncbkose** [comment]
I opened #2005, feel free to test it. Haven't tested it on Windows, but I think it should work (if you don't have too many "../"s in one path).
